kimhyojung/src/components/MovieSeats/MovieSeats.tsx

```diff
@@ -0,0 +1,262 @@
+
+    setmoviename(moviename);
+    settotalmovieprice(Number(price));
+    setmovieprice(Number(price));
+    setseatCount(0);
+    setclickedSeats([]);
+    if (typeof img === 'string') setmovieimg(img);
+  };
+
+  /*
+  좌석을 클릭했을 떄 
+  */
+  function seatbutClick(event: React.MouseEvent<HTMLButtonElement>) {
+    const currentclickedNode = event.currentTarget;
+    const currentclickedSeat = event.currentTarget.innerHTML;
+    const currentclickedNodeColor = currentclickedNode.style.backgroundColor;
+
+    //이미 예약된 좌석인 경우 예외처리
+    if (currentclickedNodeColor === NOT_POSSIBLE_SEAT_COLOR) {
+      alert(MOVIE_SEAT_ERRORMESSAGE);
+      return;
+    }
+
+    // 좌석을 눌렀을떄, 선택한 좌석을 담는 배열에 지금좌석이 있는경우(좌석을 누르고, 그 좌석을 다시 누른 경우)
+
+    /*
+      선택한 좌석을 담는 배열에 지금 좌석을 지워주고, 좌석 개수 -1
+    */
+    if (clickedSeats.includes(currentclickedSeat)) {
+      setclickedSeats(
+        clickedSeats.filter((clickedseat) => clickedseat !== currentclickedSeat)
+      );
+      currentclickedNode.style.backgroundColor = POSSIBLE_SEAT_COLOR;
+      setseatCount(seatCount - 1);
+    }
+
+    // 좌석을 한번만 누른 경우
+
+    /*
+      배열에 지금 좌석을 넣어주고, 좌석 개수 + 1
+    */
+    else {
+      setseatCount(seatCount + 1);
+      setclickedSeats([...clickedSeats, currentclickedSeat]);
+      currentclickedNode.style.backgroundColor = CURRENT_SEAT_COLOR;
+    }
+  }
+
+  /* 좌석의 개수가 바뀔떄마다, 총 내야 하는 영화 금액도 바뀜 */
+  useEffect(() => {
+    settotalmovieprice(seatCount * movieprice);
+  }, [seatCount]);
+
+  /* 영화가 바뀔때 */
+  useEffect(() => {
+    //먼저 모든 좌석색을 초기화해줌
+    const allbuttons = Array.from(document.getElementsByTagName('button'));
+    allbuttons.forEach((eachbutton) => {
+      eachbutton.style.backgroundColor = POSSIBLE_SEAT_COLOR;
+    });
+
+    //로컬 스토리지에 없는 경우
+    const isnotpossibleSeats = getlocalStorage(moviename);
+    if (isnotpossibleSeats === false) {
+      return;
+    }
+
+    // 로컬 스토리지에 있는 경우
+
+    /*
+    모든 좌석 중, 로컬 스토리지에 있는 좌석(이미 예매된 좌석)만 골라서, 그 좌석의 배경색을 바꿔줌
+    이미 선택된 좌석 배열을,  예메된 좌석들로 갱신해줌
+    */
+    [...isnotpossibleSeats].forEach((eachprevseat) => {
+      const prevseatNodes = allbuttons.filter(
+        (eachbuttonNode) => eachbuttonNode.innerHTML === eachprevseat
+      );
+
+      prevseatNodes.forEach((eachseatNode) => {
+        eachseatNode.style.backgroundColor = NOT_POSSIBLE_SEAT_COLOR;
+      });
+    });
+    setclickedSeats(isnotpossibleSeats);
+  }, [moviename]);
+
+  /*예매 버튼을 눌렀을떄 로컬스토리지에 예약된 좌석들의 값 저장 + purchase로 페이지 이동 */
+  const setLocalStorage = (event: React.MouseEvent<HTMLButtonElement>) => {
+    window.localStorage.setItem(`${moviename}`, JSON.stringify(clickedSeats));
+
+    // 좌석을 안눌렀는데, 예매버튼 누른 경우 예외처리
+    if (seatCount === 0) {
+      alert(MOVIE_EMPTY_ERRORMESSAGE);
+      return;
+    }
+    Router.push(
+      {
+        pathname: '/purchase',
+        query: {
+          moviename: moviename,
+          totalprice: totalmovieprice,
+          totalseat: seatCount,
+          movieimg: movieimg,
+        },
```

Comment:
```suggestion
        query: {
          moviename,
          totalprice,
          totalseat,
          movieimg,
        },
```

1. 이름 카멜 케이스
2. 이렇게 오브젝트에서 key 값의 이름과 변수의 명이 같으면 축약할 수 있어요

Reply:
축약해서 표현하는 것 새로 알아갑니당 👍 👍